docs/source/more_types.rst

```diff
@@ -985,3 +989,166 @@ and non-required keys, such as ``Movie`` above, will only be compatible with
+    # provides a regular bool:
+
+    @overload
+    def fetch_data(raw: bool) -> Union[bytes, str]: ...
+
+    def fetch_data(raw: bool) -> Union[bytes, str]:
+        # (Implementation is omitted)
+        pass
+
+    variable_1: Literal[True] = True
+
+    reveal_type(fetch_data(True))        # Revealed type is 'bytes'
+    reveal_type(fetch_data(False))       # Revealed type is 'str'
+    reveal_type(fetch_data(variable_1))  # Revealed type is 'bytes'
+
+    # Variables declared without annotations will continue to have an
+    # inferred type of 'bool'.
+
+    variable_2 = True
+    reveal_type(fetch_data(variable_2))  # Revealed type is 'Union[bytes, str]'
+
+Parameterizing Literals
+-----------------------
+
+Literal types may contain one or more literal bools, ints, strs, and byte
+strings. However, Literal types **cannot** contain arbitrary expressions:
+types like ``Literal[my_string.trim()]``, ``Literal[x > 3]``, or ``Literal[3j + 4]``
+are all illegal.
+
+Literals containing two or more values are equivalent to the union of those values.
+So, ``Literal[-3, b"foo", True]`` is equivalent to
+``Union[Literal[-3], Literal[b"foo"], Literal[True]]``. This makes writing
+more complex types involving Literals a little more convenient.
+
+Literal types may also contain ``None``. Mypy will treat ``Literal[None]`` as being
+equivalent to just ``None``. This means that ``Literal[4, None]``, 
+``Union[Literal[4], None]``, and ``Optional[Literal[4]]`` are all equivalent.
+
+Literals may also contain aliases of Literal types. For example, the following program
+is legal:
+
+.. code-block:: python
+
+    PrimaryColors = Literal["red", "blue", "yellow"]
+    SecondaryColors = Literal["purple", "green", "orange"]
+    AllowedColors = Literal[PrimaryColors, SecondaryColors]
+
+    def paint(color: AllowedColors) -> None: pass
+
+    paint("red")        # Type checks!
+    paint("turquoise")  # Does not type check
+
+Literals may not contain any other kind of type or expression. This means doing
+``Literal[my_instance]``, ``Literal[Any]``, ``Literal[3.14]``, or 
+``Literal[{"foo": 2, "bar": 5}]`` are all illegal.
+
+Future versions of mypy may relax some of these restrictions. For example, we
+plan on adding support for using enum values inside Literals in an upcoming release.
+
+Declaring Literal variables
+---------------------------
+
+You must explicitly add an annotation to a variable to declare that it has
+a Literal type:
+
+.. code-block:: python
+
+    a: Literal[19] = 19
+    reveal_type(a)          # Revealed type is 'Literal[19]'
+
+In order to preserve backwards-compatibility, variables without this annotation
+are **not** assumed to be Literals:
+
+.. code-block:: python
+
+    b = 19
+    reveal_type(b)          # Revealed type is 'int'
+
+If you find repeating the value of the variable in the type hint to be tedious,
+you can instead declare the variable to be :ref:`Final <final_attrs>`:
```

Comment:
"Final variable" is an kind of oxymoron. I would rather say "final name", or reformulate this otherwise.

Reply:
I decided to rephrase this to say "you can instead change the variable to be Final" (and did a similar thing below).